Packages/Core/Runtime/Extensions/IListExtensions.cs

```diff
@@ -9,17 +10,15 @@ internal static class IListExtensions
         public static T First<T>(this IList<T> list, Func<T, bool> func)
         {
             for (int i = 0; list != null && i < list.Count; ++i)
-            {
                 if (func(list[i])) return list[i];
-            }
 
             return default(T);
         }
 
         public static bool Exists<T>(this IList<T> list, Func<T, bool> func)
         {
-            var first = list.First<T>(func);
-            return first != null ? true : false;
+            var first = list.First(func);
+            return first is { };
```

Comment:
I'm not entirely sold on this one.
1. this is a pretty modern syntax (using an empty property pattern)
2. its super hard to google if one does not know what it is.
3. its less readable than !(first is null)
4. `first != null` uses the operator overloading for `!=`. `is` does not do that, though. In terms of unity this makes quite a difference when T is a UnityEngine.Object which have an overload for `!=`.
5. First<T> as it was previous uses the extension method above, using Linq.First on the other hand will **throw an exception if no match was found** - there is a FirstOrDefault, exactly doing what the First<T> overload does...

Reply:
My reason for changing it was due to the possibility of performing a null check on a value type, but if you think !(first is null) is more readable, I can swap it out to that.

**Note:** From what I can tell, the entire Exists method isn't actually used anywhere. So I think the correct solution is to remove the code entirely.